### PR TITLE
Allow archive to be run as its own module

### LINKED ISSUE #280: Adapt archive utility to run as a separate step in the pipeline
Currently, the archive utility is integrated into `run_module()`. This is problematic, because if the CSV output fails validation (either after delivery to the server, or during the [imminent validation step](https://github.com/cmu-delphi/covidcast-indicators/issues/59)) then we wind up saving files to the cache that do not actually reflect what data is in the API.

This is actively causing problems for #179, and could affect every indicator where the archiver is used in this way as soon as it fails ingestion once.

Once the validator is available, we will want to shift archiver duties to occur only after validation succeeds.

To prepare for that, we should adapt the archiver to run independently via the `delphi_utils` module. Something like
```
(env) $ python -m delphi_utils archive
```
-- the `archive` command will leave room for other utils commands should they come up.

This has the bonus of decoupling archiver code from the indicator implementations, reducing duplication, and making it easier to run indicators in development environments without needing S3 credentials.

### PR DESCRIPTION
**Pull request by sgsmob**
[edit:krivard] Fixes #280.

Archiving can now be run from the command line as a module, e.g.,
```
$ python -m delphi_utils.archive --help

usage: archive.py [-h] --archive_type {git,s3} [--indicator_prefix INDICATOR_PREFIX] [--branch_name BRANCH_NAME]
                  [--override_dirty] [--commit_partial_success] [--commit_message COMMIT_MESSAGE]

optional arguments:
  -h, --help            show this help message and exit
  --archive_type {git,s3}
                        Type of archive differ to use.
  --indicator_prefix INDICATOR_PREFIX
                        The prefix for S3 keys related to this indicator. Required for `archive_type = 's3'
  --branch_name BRANCH_NAME
                        Branch to use for `archive_type` = 'git'.
  --override_dirty      Whether to allow overwriting of untracked & uncommitted changes for `archive_type` = 'git'
  --commit_partial_success
                        Whether to still commit for `archive_type` = 'git' even if some files were not archived and staged due to
                        `override_dirty` = False.
  --commit_message COMMIT_MESSAGE
                        Commit message for `archive_type` = 'git'
```

### CODE REVIEW

**Comment by sgsmob** [comment]
See #280

File: _delphi_utils_python/delphi_utils/archive.py [hunk 1]
```diff
@@ -89,6 +94,42 @@ def diff_export_csv(
...
+               export_dir: str,
+               **kwargs):
+    """Builds and runs an ArchiveDiffer.
+
+    Parameters
+    ----------
+    archive_type: str
+        Type of ArchiveDiffer to run.  Must be one of ["git", "s3"] which correspond to `GitArchiveDiffer` and `S3ArchiveDiffer`, respectively.
+    cache_dir: str
+        The directory for storing most recent archived/uploaded CSVs to do start diffing from.
```
Comment (huisaddison) [inline _delphi_utils_python/delphi_utils/archive.py:109]:
```suggestion
         The directory for storing most recent archived/uploaded CSVs to start diffing from.
```

**Comment by huisaddison** [comment]
tl;dr : LGTM

* Read through source - code looks good - added `ArchDiffer.run()` and `run_module()` (a few diffs are line breaks to enforce character limits, which I appreciate)
* Compared the implementation of `ArchiveDiffer.run()` [[link]](https://github.com/cmu-delphi/covidcast-indicators/pull/295/files#diff-4e2435d91326a204681786df4dd349fdR270-R292) against the ArchDiffer runs in `delphi_jhu` [[link]](https://github.com/cmu-delphi/covidcast-indicators/blob/main/jhu/delphi_jhu/run.py#L120-L135) and `delphi_google_health` [[link]](https://github.com/cmu-delphi/covidcast-indicators/blob/main/google_health/delphi_google_health/run.py#L93-L107).  Looks like a straightforward abstraction of the functionality to me.
* Unit tests look good to me (also looks like @sgsmob deduped some local constants into global constants :) ).  Checked out the PR locally, installed module, ran unit tests, everything passes.

@krivard is there anything else you want me to look into for this code review?  For @sgsmob 's context - I am not a professional software engineer, and I haven't been heavily involved in engineering since April-ish.  I wrote a bunch of initial indicator code, then switched to modeling for the summer, and then I came back to eng because the structure allows me to get Phd student work done in parallel :P  So my main strength is probably explaining why there are certain hacks in legacy code (and spending time fixing them to pay down our tech debt!)

**Comment by krivard** [comment]
Nope! Just getting some additional eyeballs on the code. Thank you!
